webapi/src/main/twirl/queries/sparql/v1/deleteResource.scala.txt

```diff
@@ -0,0 +1,91 @@
+ * Tobias Schweizer, André Kilchenmann, and André Fatton.
+ *
+ * This file is part of Knora.
+ *
+ * Knora is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ *@
+
+@import org.knora.webapi._
+
+@**
+ * Marks a resource as deleted.
+ *
+ * @param dataNamedGraph the named graph to update.
+ * @param triplestore the name of the triplestore being used.
+ * @param resourceIri the IRI of the resource to be marked as deleted.
+ * @param maybeDeleteComment an optional comment explaining why the value is being deleted.
+ *@
+@(dataNamedGraph: IRI,
+  triplestore: String,
+  resourceIri: IRI,
+  maybeDeleteComment: Option[String])
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+
+DELETE {
+    GRAPH ?dataNamedGraph {
+        @* Delete the resource's last modification date so we can update it. *@
+        ?resource knora-base:lastModificationDate ?resourceLastModificationDate .
+
+        @* Delete the resource's isDeleted flag so we can update it. *@
+        ?resource knora-base:isDeleted false .
+    }
+} INSERT {
+    GRAPH ?dataNamedGraph {
+        @* Mark the resource as deleted. *@
+        ?resource knora-base:isDeleted true ;
+            knora-base:deleteDate ?currentTime .
+
+        @* Add the delete comment, if any. *@
+        @maybeDeleteComment match {
+            case Some(comment) => {
+
+                ?resource knora-base:deleteComment """@comment""" .
+
+            }
+
+            case None => {}
+        }
+
+        @* Update the resource's last modification date. *@
+        ?resource knora-base:lastModificationDate ?currentTime .
+    }
+}
+@* Ensure that inference is not used in the WHERE clause of this update. *@
+@if(triplestore.startsWith("graphdb")) {
+    USING <http://www.ontotext.com/explicit>
+}
+WHERE {
+    BIND(IRI("@dataNamedGraph") AS ?dataNamedGraph)
+    BIND(IRI("@resourceIri") AS ?resource)
+    BIND(NOW() AS ?currentTime)
+
+    @* Make sure the resource is really a resource and isn't marked as deleted. *@
+
+    ?resource rdf:type ?resourceClass ;
+        knora-base:isDeleted false .
+    ?resourceClass rdfs:subClassOf+ knora-base:Resource .
```

Comment:
The use of `rdfs:subClassOf+ knora-base:Resource` means that this works for sublasses of knora-base:Resource, but not for a knora-base:Resource itself.

However, from the GUI it is possible to create a knora-base:Resource but this fails when making a get request providing the new resource's IRI.

So we have to decide whether knora-base:Resource can be instantiated or only sublasses of it. If yes, we should use the operator *. And we would have to do that in a consistent manner (in every SPARQL template dealing with resources)